prow/plugins/commands/handlers/handlers.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"fmt"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/plugins/commands"
+)
+
+type labelClient interface {
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+	AddLabel(org, repo string, number int, label string) error
+	RemoveLabel(org, repo string, number int, label string) error
+}
+
+// EnsureLabel adds the specified label if it is not already present.
+func EnsureLabel(label string) commands.MatchHandler {
+	return func(ctx *commands.Context) error {
+		return doEnsureLabel(ctx.Client.GitHubClient, ctx.Event, label)
+	}
+}
+
+func doEnsureLabel(client labelClient, e *github.GenericCommentEvent, label string) error {
+	org := e.Repo.Owner.Login
+	repo := e.Repo.Name
+	number := e.Number
+
+	currentLabels, err := client.GetIssueLabels(org, repo, number)
```

Comment:
What does the API do if we call `AddLabel()` and it already exists? Can we save a token here?

Reply:
It returns 200 either way.  Whether we can save an API token depends on if there is a warm cache entry for the label list and if the label is already present or not. This could actually use fewer tokens, see the comment on `RemoveLabel`.

It is unclear which of these forms is cheaper in practice with the GitHub cache. If we want to minimize API token usage we should probably gather some data and optimize this in a follow up PR.